src/Microsoft.Graph.Cli.Core/Http/HttpHeadersHandler.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Graph.Cli.Core.Http;
+
+public class HttpHeadersHandler : DelegatingHandler
+{
+    protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        foreach (var headerItem in HeadersStore.Instance.Headers)
+        {
+            request.Headers.Add(headerItem.Key, headerItem.Value);
```

Comment:
This operation can throw an `InvalidOperationException` e.g when the header is for the content and not for the request message. We should probably use `TryAddWithoutValidation` and check the return value to avoid this.
Similar to the code at https://github.com/microsoft/kiota-http-dotnet/blob/848ceaf64bb7d5387b6435f6b679593adb3550a9/src/HttpClientRequestAdapter.cs#L510

Reply:
🤔`TryAddWithoutValidation` has a different signature. I can use a try and catch.